### PR TITLE
ci: Safe commenting on PRs

### PR DESCRIPTION
**Pull request by larseggert**
This generalizes and refactors what the QNS workflow did.

### CODE REVIEW

File: .github/actions/quic-interop-runner/action.yml [hunk 1]
```diff
@@ -88,24 +88,20 @@ runs:
...
+        cat quic-interop-runner/summary >> comment
+        echo '```' >> comment
+        echo '' >> comment
+      shell: bash
 
-    - name: Upload comment data
-      uses: actions/upload-artifact@v4
-      if: github.event_name == 'pull_request'
+    - name: Export PR comment data
+      if: ${{ github.event_name == 'pull_request' }}
```
Comment (mxinden) [inline .github/actions/quic-interop-runner/action.yml:102]: Each step in `pr-comment-data-export` has this `if` already. Can we remove it either here or in the action?

**Review by mxinden — APPROVED** [review]
Thank you for pulling this into separate actions! Overall looks good to me. Couple of nits / questions.

File: .github/workflows/qns-comment.yml [hunk 2]
```diff
@@ -17,40 +17,8 @@ jobs:
     permissions:
       pull-requests: write
     runs-on: ubuntu-latest
-    if: >
-      github.event.workflow_run.event == 'pull_request' &&
-      github.event.workflow_run.conclusion == 'failure'
+    if: github.event.workflow_run.event == 'pull_request'
```
Comment (mxinden) [inline .github/workflows/qns-comment.yml:20]: In the case of qns, the comment should still only be printed on `failure`, right?

Comment (larseggert) [reply .github/workflows/qns-comment.yml:20]: Yes, this needs to be restored - I was just trying to make it generate any output.

**Pull request merged**